swiftpkg/internal/swiftpkg_build_files.bzl

```diff
@@ -219,81 +220,83 @@ def _swift_compiler_plugin_from_target(target, attrs):
 
 # MARK: - Clang Targets
 
+def _c_child_library(
+        repository_ctx,
+        name,
+        attrs,
+        rule_kind,
+        srcs,
+        language_standard = None,
+        res_copts = None):
+    child_attrs = dict(**attrs)
+    child_copts = list(attrs.get("copts", []))
+    if res_copts:
+        child_copts.extend(res_copts)
+    child_attrs["srcs"] = lists.flatten([srcs, attrs.get("srcs", [])])
+    if language_standard:
+        child_copts.append("-std={}".format(language_standard))
+    child_attrs["copts"] = child_copts
+    return build_decls.new(
+        rule_kind,
+        name,
+        attrs = _starlarkify_clang_attrs(repository_ctx, child_attrs),
```

Comment:
I think we need to set the `@build_bazel_rules_swift//swift:no_module` aspect hint on the children. And then we need to set all of the headers on the "uber" target (so stop omitting `hdrs` and `textual_hdrs`) and probably stop setting `hdrs` and `textual_hdrs` on the child targets, so the generated modulemap for the uber target represents all of the headers.

This change will make it so these child targets only compile source, but don't propagate headers or modulemaps. Then the "uber" target does what the old version did, propagating headers and modulemap.

Reply:
I took a stab at adding the aspect hint and consolidating the headers. Things break. So, I checked the generated files. The look right to me. WDYT?

## BUILD.bazel

```bazel
objc_library(
    name = "GoogleUtilities-AppDelegateSwizzler.rspm",
    copts = [
        "-fblocks",
        "-fobjc-arc",
        "-fPIC",
        "-DSWIFT_PACKAGE=1",
        "-Iexternal/rules_swift_package_manager~~swift_deps~swiftpkg_googleutilities/GoogleUtilities/AppDelegateSwizzler/Internal",
        "-Iexternal/rules_swift_package_manager~~swift_deps~swiftpkg_googleutilities/GoogleUtilities/Common",
        "-Iexternal/rules_swift_package_manager~~swift_deps~swiftpkg_googleutilities",
        "-fmodule-name=GoogleUtilities_AppDelegateSwizzler",
    ],
    deps = [":GoogleUtilities-AppDelegateSwizzler.rspm_objc"],
    enable_modules = True,
    includes = ["GoogleUtilities/AppDelegateSwizzler/Public"],
    module_name = "GoogleUtilities_AppDelegateSwizzler",
    sdk_frameworks = select({
        "@rules_swift_package_manager//config_settings/spm/platform:ios": [
            "Foundation",
            "UIKit",
        ],
        "@rules_swift_package_manager//config_settings/spm/platform:macos": [
            "AppKit",
            "Foundation",
        ],
        "@rules_swift_package_manager//config_settings/spm/platform:tvos": [
            "Foundation",
            "UIKit",
        ],
        "@rules_swift_package_manager//config_settings/spm/platform:watchos": [
            "Foundation",
            "UIKit",
            "WatchKit",
        ],
        "//conditions:default": [],
    }),
    visibility = ["//:__subpackages__"],
)

objc_library(
    name = "GoogleUtilities-AppDelegateSwizzler.rspm_objc",
    aspect_hints = ["@build_bazel_rules_swift//swift:no_module"],
    copts = [
        "-fblocks",
        "-fobjc-arc",
        "-fPIC",
        "-DSWIFT_PACKAGE=1",
        "-Iexternal/rules_swift_package_manager~~swift_deps~swiftpkg_googleutilities/GoogleUtilities/AppDelegateSwizzler/Internal",
        "-Iexternal/rules_swift_package_manager~~swift_deps~swiftpkg_googleutilities/GoogleUtilities/Common",
        "-Iexternal/rules_swift_package_manager~~swift_deps~swiftpkg_googleutilities",
        "-std=c99",
    ],
    deps = [
        "@swiftpkg_googleutilities//:GoogleUtilities-Environment.rspm",
        "@swiftpkg_googleutilities//:GoogleUtilities-Environment.rspm_modulemap",
        "@swiftpkg_googleutilities//:GoogleUtilities-Logger.rspm",
        "@swiftpkg_googleutilities//:GoogleUtilities-Logger.rspm_modulemap",
        "@swiftpkg_googleutilities//:GoogleUtilities-Network.rspm",
        "@swiftpkg_googleutilities//:GoogleUtilities-Network.rspm_modulemap",
    ],
    enable_modules = True,
    hdrs = [
        "GoogleUtilities/AppDelegateSwizzler/Public/GoogleUtilities/GULAppDelegateSwizzler.h",
        "GoogleUtilities/AppDelegateSwizzler/Public/GoogleUtilities/GULApplication.h",
        "GoogleUtilities/AppDelegateSwizzler/Public/GoogleUtilities/GULSceneDelegateSwizzler.h",
    ],
    includes = ["GoogleUtilities/AppDelegateSwizzler/Public"],
    sdk_frameworks = select({
        "@rules_swift_package_manager//config_settings/spm/platform:ios": [
            "Foundation",
            "UIKit",
        ],
        "@rules_swift_package_manager//config_settings/spm/platform:macos": [
            "AppKit",
            "Foundation",
        ],
        "@rules_swift_package_manager//config_settings/spm/platform:tvos": [
            "Foundation",
            "UIKit",
        ],
        "@rules_swift_package_manager//config_settings/spm/platform:watchos": [
            "Foundation",
            "UIKit",
            "WatchKit",
        ],
        "//conditions:default": [],
    }),
    srcs = [
        "GoogleUtilities/AppDelegateSwizzler/GULAppDelegateSwizzler.m",
        "GoogleUtilities/AppDelegateSwizzler/GULSceneDelegateSwizzler.m",
        "GoogleUtilities/AppDelegateSwizzler/Internal/GULAppDelegateSwizzler_Private.h",
        "GoogleUtilities/AppDelegateSwizzler/Internal/GULSceneDelegateSwizzler_Private.h",
        "GoogleUtilities/Common/GULLoggerCodes.h",
        "third_party/IsAppEncrypted/Public/IsAppEncrypted.h",
        "GoogleUtilities/UserDefaults/Public/GoogleUtilities/GULUserDefaults.h",
        "GoogleUtilities/MethodSwizzler/Public/GoogleUtilities/GULSwizzler.h",
        "GoogleUtilities/MethodSwizzler/Public/GoogleUtilities/GULOriginalIMPConvenienceMacros.h",
        "GoogleUtilities/Logger/Public/GoogleUtilities/GULLoggerLevel.h",
        "GoogleUtilities/Logger/Public/GoogleUtilities/GULLogger.h",
        "GoogleUtilities/Network/GULNetworkInternal.h",
        "GoogleUtilities/Network/Public/GoogleUtilities/GULNetworkMessageCode.h",
        "GoogleUtilities/Network/Public/GoogleUtilities/GULMutableDictionary.h",
        "GoogleUtilities/Network/Public/GoogleUtilities/GULNetworkURLSession.h",
        "GoogleUtilities/Network/Public/GoogleUtilities/GULNetworkLoggerProtocol.h",
        "GoogleUtilities/Network/Public/GoogleUtilities/GULNetwork.h",
        "GoogleUtilities/Network/Public/GoogleUtilities/GULNetworkConstants.h",
        "GoogleUtilities/Tests/Unit/Network/third_party/GTMHTTPServer.h",
        "GoogleUtilities/Tests/Unit/Utils/GULTestKeychain.h",
        "GoogleUtilities/Tests/Unit/Shared/URLSession/FIRURLSessionOCMockStub.h",
        "GoogleUtilities/Tests/Unit/Environment/Sources/GULHeartbeatDateStorageV7.4.0.h",
        "GoogleUtilities/Tests/Unit/Environment/Sources/GULHeartbeatDateStorageV7.3.1.h",
        "GoogleUtilities/Reachability/GULReachabilityMessageCode.h",
        "GoogleUtilities/Reachability/GULReachabilityChecker+Internal.h",
        "GoogleUtilities/Reachability/Public/GoogleUtilities/GULReachabilityChecker.h",
        "GoogleUtilities/Environment/Public/GoogleUtilities/GULHeartbeatDateStorable.h",
        "GoogleUtilities/Environment/Public/GoogleUtilities/GULAppEnvironmentUtil.h",
        "GoogleUtilities/Environment/Public/GoogleUtilities/GULHeartbeatDateStorage.h",
        "GoogleUtilities/Environment/Public/GoogleUtilities/GULURLSessionDataResponse.h",
        "GoogleUtilities/Environment/Public/GoogleUtilities/GULKeychainStorage.h",
        "GoogleUtilities/Environment/Public/GoogleUtilities/GULHeartbeatDateStorageUserDefaults.h",
        "GoogleUtilities/Environment/Public/GoogleUtilities/NSURLSession+GULPromises.h",
        "GoogleUtilities/Environment/Public/GoogleUtilities/GULNetworkInfo.h",
        "GoogleUtilities/Environment/Public/GoogleUtilities/GULSecureCoding.h",
        "GoogleUtilities/Environment/Public/GoogleUtilities/GULKeychainUtils.h",
        "GoogleUtilities/SwizzlerTestHelpers/GULRuntimeClassDiff.h",
        "GoogleUtilities/SwizzlerTestHelpers/GULRuntimeSnapshot.h",
        "GoogleUtilities/SwizzlerTestHelpers/GULRuntimeClassSnapshot.h",
        "GoogleUtilities/SwizzlerTestHelpers/GULRuntimeStateHelper.h",
        "GoogleUtilities/SwizzlerTestHelpers/Public/GoogleUtilities/GULSwizzler+Unswizzle.h",
        "GoogleUtilities/SwizzlerTestHelpers/GULSwizzlingCache_Private.h",
        "GoogleUtilities/SwizzlerTestHelpers/GULSwizzlingCache.h",
        "GoogleUtilities/SwizzlerTestHelpers/GULRuntimeDiff.h",
        "GoogleUtilities/SwizzlerTestHelpers/GULProxy.h",
        "GoogleUtilities/ISASwizzler/Public/GoogleUtilities/GULObjectSwizzler.h",
        "GoogleUtilities/ISASwizzler/Public/GoogleUtilities/GULSwizzledObject.h",
        "GoogleUtilities/ISASwizzler/GULObjectSwizzler+Internal.h",
        "GoogleUtilities/NSData+zlib/Public/GoogleUtilities/GULNSData+zlib.h",
    ],
    textual_hdrs = [
        "GoogleUtilities/AppDelegateSwizzler/GULAppDelegateSwizzler.m",
        "GoogleUtilities/AppDelegateSwizzler/GULSceneDelegateSwizzler.m",
    ],
    visibility = ["//:__subpackages__"],
)
```

## module.modulemap

```
module "GoogleUtilities_AppDelegateSwizzler" {

    header "../../../../../../external/rules_swift_package_manager~~swift_deps~swiftpkg_googleutilities/GoogleUtilities/AppDelegateSwizzler/Public/GoogleUtilities/GULAppDelegateSwizzler.h"
    header "../../../../../../external/rules_swift_package_manager~~swift_deps~swiftpkg_googleutilities/GoogleUtilities/AppDelegateSwizzler/Public/GoogleUtilities/GULApplication.h"
    header "../../../../../../external/rules_swift_package_manager~~swift_deps~swiftpkg_googleutilities/GoogleUtilities/AppDelegateSwizzler/Public/GoogleUtilities/GULSceneDelegateSwizzler.h"

    use "GoogleUtilities_Environment"
    use "GoogleUtilities_Logger"
    use "GoogleUtilities_Network"

    export *
}
```